### DISCUSSION TITLE
Error " unable to save state file: a session with PID 25992 is already running on platform "cluster" " with odo v3.11.0 using Openshift local

### DISCUSSION THREAD

**Discussion by jochenr** [redhat-developer/odo#6958] [Q&A] [answered]
Hello, 

using odo (from vscode's openshift tooling) I sporadically/very often get this error:

Starting odo dev --debug
  __
 /  \__     Developing using the "quarkus-camel-xxx" Devfile
 \__/  \    Namespace: xxx-dev
 /  \__/    odo version: v3.11.0
 \__/

- Running on the cluster in Dev mode
 X  unable to save state file: a session with PID 25992 is already running on platform "cluster"

Press any key to close this terminal

Environment is
  Openshift local 
  CRC version: 2.20.0+f3a947
  OpenShift version: 4.13.0

running on Windows 10

I am not able to finde the cause of the problem.

I'd appreciate some help.

Thank you,
Jochen

**Comment by rm3l** [comment]
Hi @jochenr,

From the error message, it looks like there is already an "odo dev" session running on your system for this component. `odo` only supports one running Dev Session per platform type (`cluster` or `podman`).

Can you please check if there is any running process with the PID reported by `odo` (`25992` from the output you shared)?
Maybe we have a bug in the way we detect already running processes on Windows.

A workaround can also be to restart the operating system.

**Reply by jochenr** [reply]
Hi @rm3l ,

thank you for the hint.
This morning, when I experienced the issue again, I took a look on the windows task manager and didn't find the PID.
Then I used "get-process" in powershell and found it.

The strange thing is that the PID belonged to a completely other program I was running (https://conemu.github.io/   a terminal window  running my AMQ Artemis instance).

After killing this process, ODO worked fine.

But why does ODO try to use a PID that is already used by another program?

**Reply by rm3l** [reply, marked as answer]
Thanks for the detailed information!

> But why does ODO try to use a PID that is already used by another program?

When starting, `odo dev` stores its own PID into an internal `.odo/devstate*` file (in the project working dir). Then if you try to run `odo dev` again against a same platform, it uses this file and checks whether there is already a running process with that PID. This file should be deleted when `odo dev` is stopped.
After discussing this with the rest of the team, the problem we suspect here is that the `odo dev` session might not be properly stopped by the VSCode Plugin on Windows, and the previous PID recorded was assigned to a different process by the operating system (but we'll need to confirm this and report the issue). And we plan to create an issue to make `odo delete component` able to properly delete all these internal state files.
Meanwhile, I think a simpler workaround for you can be to manually delete all the `.odo/devstate*` files from the project working directory (where your `devfile.yaml` is located) before starting `odo`.

**Reply by jochenr** [reply]
Thank you very much!👍

The workaround helped😃
